be/src/exec/pipeline/scan/morsel.cpp

```diff
@@ -759,7 +759,7 @@ Status LogicalSplitMorselQueue::_init_tablet() {
     }
 
     _largest_rowset = _find_largest_rowset(_tablet_rowsets[_tablet_idx]);
-    if (_largest_rowset == nullptr || _largest_rowset->num_rows() == 0) {
+    if (_largest_rowset == nullptr || _largest_rowset->num_rows() == 0 || _tablets[_tablet_idx]->num_rows() == 0) {
```

Comment:
this change looks good to me, but i have another question.
If `_tablets[_tablet_idx]->num_rows()` is super small like 1 when there are many deletes, will `_splitted_scan_rows * _segment_group->num_blocks() / _tablets[_tablet_idx]->num_rows();` be a super large number? Then it will cause `_sample_splitted_scan_blocks` to be too big, will it influence the performance?

Reply:
Good suggestions. Use `std::max({1, _tablets[_tablet_idx]->num_rows(), _largest_rowset->num_rows(), _segment_group->num_rows()})` instead of `_tablets[_tablet_idx]->num_rows()`.